### ISSUE TITLE
Effective storage with custom list implementations

### ISSUE THREAD

**Issue by dlux-google** [google/built_collection.dart#211]
It would be great to have `BuiltList` to support a custom backing store for specialied lists, like the ones in `dart:typed_data`.

I'm thinking of passing a factory function to `ListBuilder` that creates the backing store before copying the values over. Something like typedef `CreateListFunction<E> = List<E> Function();`

**Comment by davidmorgan** [comment]
I'm wondering if we should go further and have a separate implementation / codepath for these, so that we can use the typed data in the builders, too. That will take some work because the typed lists are fixed length, but it seems like it should save a lot of memory--WDYT?

**Comment by dlux-google** [comment]
Yes, separate implementation looks like a good idea to me, dealing with fixed length arrays as a backing store will likely increase complexity quite a bit.
